Source/Scene/ImplicitTileset.js

```diff
@@ -0,0 +1,244 @@
+import defined from "../Core/defined.js";
+import Resource from "../Core/Resource.js";
+import RuntimeError from "../Core/RuntimeError.js";
+import has3DTilesExtension from "./has3DTilesExtension.js";
+import ImplicitSubdivisionScheme from "./ImplicitSubdivisionScheme.js";
+
+/**
+ * An ImplicitTileset is a simple struct that stores information about the
+ * structure of a single implicit tileset. This includes template URIs for
+ * locating resources, details from the implicit root tile (bounding volume,
+ * geometricError, etc.), and details about the subtrees (e.g. subtreeLevels,
+ * subdivisionScheme).
+ *
+ * @alias ImplicitTileset
+ * @constructor
+ *
+ * @param {Cesium3DTileset} tileset The tileset this implicit tileset belongs to. Used for accessing metadata properties
+ * @param {Resource} baseResource The base resource for the tileset
+ * @param {Object} tileJson The JSON header of the tile with the 3DTILES_implicit_tiling extension.
+ * @private
+ * @experimental This feature is using part of the 3D Tiles spec that is not final and is subject to change without Cesium's standard deprecation policy.
+ */
+export default function ImplicitTileset(tileset, baseResource, tileJson) {
+  var extension = tileJson.extensions["3DTILES_implicit_tiling"];
+  //>>includeStart('debug', pragmas.debug);
+  Check.typeOf.object(
+    'tileJson.extensions["3DTILES_implicit_tiling"]',
+    extension
+  );
+  //>>includeEnd('debug');
+
+  /**
+   * A reference to the tileset to which this implicit tileset belongs.
+   *
+   * @type {Cesium3DTileset}
+   * @readonly
+   * @private
+   */
+  this.tileset = tileset;
+
+  /**
+   * The base resource for the tileset. This is stored here as it is needed
+   * later when expanding Implicit3DTileContents so tile URLs are relative
+   * to the tileset, not the subtree file.
+   *
+   * @type {Resource}
+   * @readonly
+   * @private
+   */
+  this.baseResource = baseResource;
+
+  /**
+   * The geometric error of the root tile
+   *
+   * @type {Number}
+   * @readonly
+   * @private
+   */
+  this.geometricError = tileJson.geometricError;
+
+  if (
+    !defined(tileJson.boundingVolume.box) &&
+    !defined(tileJson.boundingVolume.region)
+  ) {
+    throw new RuntimeError(
+      "Only box and region are supported for implicit tiling"
+    );
+  }
+
+  /**
+   * The JSON representation of a bounding volume. This is either a box or a
+   * region.
+   *
+   * @type {Object}
+   * @readonly
+   * @private
+   */
+  this.boundingVolume = tileJson.boundingVolume;
+
+  /**
+   * The refine strategy as a string, either 'ADD' or 'REPLACE'
+   *
+   * @type {String}
+   * @readonly
+   * @private
+   */
+  this.refine = tileJson.refine;
+
+  /**
+   * Template URI for the subtree resources, e.g.
+   * <code>https://example.com/{level}/{x}/{y}.subtree</code>
+   *
+   * @type {Resource}
+   * @readonly
+   * @private
+   */
+  this.subtreeUriTemplate = new Resource({ url: extension.subtrees.uri });
```

Comment:
It's interesting that this creates a new relative resource instead of deriving from the `baseResource` that is passed in. I assume it has something to do with how `Cesium3DTileset` works.

To get a resource with a full url, I had to do something like:
```
var subtreeRelative = implicitTileset.subtreeUriTemplate.getDerivedResource(
{
    templateValues: subtreeCoord.getTemplateValues()
});
var subtreeResource = implicitTileset.baseResource.getDerivedResource(
    url: subtreeRelative.url
});
```

But it would be nice if it were:
```
var subtreeResource = implicitTileset.subtreeUriTemplate.getDerivedResource(
    templateValues: subtreeCoord.getTemplateValues()
});
```

Reply:
Took me a while to remember, but I think I know why I did it this way: since this implementation was a transcoder, I don't actually need an absolute URI, a relative one is fine. E.g. the template URI is `{level}/{x}/{y}.subtree`. Once transcoded it becomes `0/0/0.subtree`, not `https://example.com/path/to/0/0/0.subtree` in the generated `content.uri` field.

What's your use case?

Reply:
I talked with @IanLilleyT about this today, we realized that there's another thing lurking here: we should try 3d-tiles-next out with resources in ion to see if they work. `IonResource` does more than just format the URL (it handles the access tokens) so there could be other problems lurking here...

Reply:
Oh forgot to mention, the design we plan on doing is:

```
ImplicitTileset.prototype.deriveSubtreeResource(implicitCoordinates: ImplicitTileCoordinates, relativePath: Boolean) -> Resource

ImplicitTileset.prototype.deriveContentResource(implicitCoordinates: ImplicitTileCoordinates, relativePath: Boolean, index=0) -> Resource
```

Reply:
The current implementation does indeed work with `IonResource`

Though as for the derive resource idea... @IanLilleyT could that be a follow-up PR like we did with #9531? this PR is already huge as it is.